openapi3.yaml

```diff
@@ -1,88 +1,488 @@
+          geometry_type:
+            $ref: '#/components/schemas/geometryType'
+          layer_id:
+            $ref: '#/components/schemas/layerId'
+          is_hidden:
+            $ref: '#/components/schemas/isHidden'
+          exclusive_from:
+            $ref: '#/components/schemas/replicaDate'
+          to:
+            $ref: '#/components/schemas/replicaDate'
+        additionalProperties: false
+        minProperties: 1
+    replicaType:
+      in: query
+      name: replica_type
+      schema:
+        $ref: '#/components/schemas/replicaType'
+    layerId:
+      in: query
+      name: layer_id
+      schema:
+        $ref: '#/components/schemas/layerId'
+  responses:
+    BadRequest:
+      description: Bad Request
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/error'
+    NotFound:
+      description: Not Found
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/error'
+    Unauthorized:
+      description: Unauthorized Error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/error'
+    Conflict:
+      description: Conflict
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/error'
+    UnexpectedError:
+      description: Unexpected Error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/error'
   schemas:
+    geometryType:
+      type: string
+      enum:
+        - point
+        - linestring
+        - polygon
+    replicaType:
+      type: string
+      enum:
+        - snapshot
+        - delta
+    replicaDate:
+      type: string
+      format: date-time
+    replicaId:
+      type: string
+      format: uuid
+    fileId:
+      type: string
+      format: uuid
+    layerId:
+      type: number
+    layerName:
+      type: string
+    isHidden:
+      type: boolean
     error:
       type: object
       required:
         - message
       properties:
         message:
           type: string
-    resource:
+    baseReplica:
       type: object
-      required:
-        - id
-        - name
-        - description
       properties:
-        id:
+        timestamp:
+          $ref: '#/components/schemas/replicaDate'
+        layerId:
           type: number
-          format: int64
-        name:
-          type: string
-        description:
-          type: string
-    anotherResource:
-      type: object
-      required:
-        - kind
-        - isAlive
-        - description
-      properties:
-        kind:
+          format: int32
+        geometryType:
+          $ref: '#/components/schemas/geometryType'
+        bucketName:
           type: string
-        isAlive:
-          type: boolean
+          minLength: 3
+          maxLength: 63
+          writeOnly: true
+        replicaType:
+          $ref: '#/components/schemas/replicaType'
+        url:
```

Comment:
the spec says the property name is `url` in singular while the service returns `urls` as property name, since it is an array of urls, I think it is better if we stick to `urls` instead of url.

Reply:
done.